### ISSUE TITLE
Implement the "View Profile" page for users

### ISSUE THREAD

**Issue by michaelsena** [3box/3box-dapp#195]
# Description
As a 3Box user managing my presence to the web3 world on 3Box, I want a way to see how others can see my profile and confirm the amount of information that I am disclosing.

## Details
- this should live at the user's unique URL: https://github.com/3box/3box-dapp/issues/16

# Design Spec
Something like this:
![Profile View Page.png](https://images.zenhubusercontent.com/5b1e6edb4b5806bc2bd704db/cac68b5f-189f-48db-9615-56590429b3ff)

**Comment by oznekenzo** [comment]
Before there was essentially a two by two matrix by which a user enters the dapp:
1) Is user logged in (T/F)  
2) did user arrive to app on protected routes (T/F)

Shareable profiles adds a third layer
3) did user arrive to app on public profile

We have to handle this third case in different ways depending on if the user landed there *after* going through the flow of a user thats logged in vs one that isn't.

TLDR
The dapp logic handling openBox, web3 address & consent checks, getting user profile data all has to get reshuffled to accommodate for the third possibility of landing on a shared profile.  
Going to take some time to get it worked out and bullet proof

**Comment by oznekenzo** [comment]
My comments in discord:
get and render public profile

show the nav we show on the landing page, or a special nav we havent specced yet (because we don't know yet if the user is logged in or not, and don't have account details for that user like a profile picture).  We show this nav until we have successfully openBoxed and retrieved user data in the background.

request "Connect" - initiate access to ethereum account right when user lands on page, which will prompt a metamask dialogue, otherwise we cannot initiate a loading of the dapp in the background

check eth network if logged in

if user is signed into their web3 wallet and is logged in to 3box, get box
if user is not signed into their web3 wallet, show a special, smaller modal to sign in to their web3 wallet that doesn't obstruct public profile
if user is signed into their web3 wallet but is not logged in to 3box,  show a special, smaller modal to sign in to 3Box that doesn't obstruct public profile, this might include a button on that modal that can begin the sign in process that can be done in the background while user is on public profile

edge cases: moving to landing page during lazy load,

**Comment by oznekenzo** [comment]
From @michaelsena on discord:
Routes:
- remove /user/ and update to this pattern:
- General route to someone’s profile:
    - 3box.io/0xa8ee0babe72cd9a80ae45dd74cd3eae7a82fd5d1
- If you are signed in with that same ethereum address, then take the user directly to the route they enter, otherwise default send them to /activity. If they enter the general route, take them to /activity. 
    - 3box.io/0xa8ee0babe72cd9a80ae45dd74cd3eae7a82fd5d1/activity 
    - 3box.io/0xa8ee0babe72cd9a80ae45dd74cd3eae7a82fd5d1/about 
    -  3box.io/0xa8ee0babe72cd9a80ae45dd74cd3eae7a82fd5d1/edit
- This makes it easy that if someone is on their profile and just highlights the URL in the URLbar, and shares it on twitter (as I would if I wanted to share my profile), we can just direct the person who clicks that link to the base path, if they are: 1) not signed in, or 2) are signed in but are not the owner of that address

Share + View Buttons:
- If we do this ^, we can remove the “share” button from the My Profile screens, since people will likely just copy their own URL anyway.
- Then the question is how do we handle the “View” button - which we may be able to remove now too, since the user can view their own profile if they hit their base path or any other path, while signed out. (This is how twitter does it, for example)

General Route View:
- Should not have separate Activity and Details sections, but instead should be one integrated view, as described in the mockup. This will create an “at a glance” view of someones complete profile without breaking information up.
- Show an altered status view, as in the mockup, again to create difference

Signed Out NavBar:
- Let's revisit after we make the other changes

—
My Details Page:
- Remove “Edit” button from every category, except the top right corner of the element
- Make that “Edit” always visible and #1168df blue
- Make Twitter, Github, and URL links clickable
